src/js/components/FileInput/FileInput.js

```diff
@@ -158,32 +160,104 @@ const FileInput = forwardRef(
 
     return (
-      <ContentsBox
-        theme={theme}
-        flex={false}
-        disabled={disabled}
-        background={mergeTheme('background', 'color')}
-        border={!plain ? mergeTheme('border', 'side') : undefined}
-        margin={mergeTheme('margin')}
-        pad={mergeTheme('pad')}
-        round={mergeTheme('round', 'size')}
-        align={files.length ? 'stretch' : 'center'}
-        justify="center"
-        hover={hover}
-        onMouseOver={disabled ? undefined : () => setHover(true)}
-        onMouseOut={disabled ? undefined : () => setHover(false)}
-        dragOver={dragOver}
-      >
-        {(!files.length || files.length > 1) && (
-          <Box
-            align="center"
-            fill="horizontal"
-            direction="row"
-            justify="between"
-          >
-            {files.length <= aggregateThreshold && (
-              <>
-                <Message {...theme.fileInput.message}>{message}</Message>
+      <>
+        <ContentsBox
+          theme={theme}
+          flex={false}
+          disabled={disabled}
+          background={mergeTheme('background', 'color')}
+          border={!plain ? mergeTheme('border', 'side') : undefined}
+          margin={mergeTheme('margin')}
+          pad={mergeTheme('pad')}
+          round={mergeTheme('round', 'size')}
+          align={files.length ? 'stretch' : 'center'}
+          justify="center"
+          hover={hover}
+          onMouseOver={disabled ? undefined : () => setHover(true)}
+          onMouseOut={disabled ? undefined : () => setHover(false)}
+          dragOver={dragOver}
+        >
+          {(!files.length || files.length > 1) && (
+            <Box
+              align="center"
+              fill="horizontal"
+              direction="row"
+              justify="between"
+            >
+              {files.length <= aggregateThreshold && (
+                <>
+                  <Message {...theme.fileInput.message}>{message}</Message>
+                  <Keyboard
+                    onSpace={(event) => {
+                      if (controlRef.current === event.target)
+                        inputRef.current.click();
+                    }}
+                    onEnter={(event) => {
+                      if (controlRef.current === event.target)
+                        inputRef.current.click();
+                    }}
+                  >
+                    {theme.fileInput.button ? (
+                      <Button
+                        ref={controlRef}
+                        kind={theme.fileInput.button}
+                        label={format({
+                          id: 'fileInput.browse',
+                          messages,
+                        })}
+                        onClick={() => {
+                          inputRef.current.click();
+                          inputRef.current.focus();
+                        }}
+                        disabled={max === files.length}
```

Comment:
I think this is kinda accidentally working when files.length = 0 and max isn't defined or is basically coming out `false`.

Reply:
Can you please suggest a better solution ?